geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySender.java

```diff
@@ -1034,9 +1036,17 @@ public void distribute(EnumListenerEvent operation, EntryEventImpl event,
       // If this gateway is not running, return
       if (!isRunning()) {
         if (this.isPrimary()) {
-          tmpDroppedEvents.add(clonedEvent);
-          if (isDebugEnabled) {
-            logger.debug("add to tmpDroppedEvents for evnet {}", clonedEvent);
+          if (this.startTime < 0
+              && System.currentTimeMillis() + this.startTime < maxWaitTimeForSenderToStart) {
+            tmpDroppedEvents.add(clonedEvent);
+            if (isDebugEnabled) {
+              logger.debug("add to tmpDroppedEvents for evnet {}", clonedEvent);
+            }
+          } else {
+            if (isDebugEnabled) {
+              logger.debug("The sender is stopped. Not to add to tmpDroppedEvents for event {}",
```

Comment:
This might be worded better as "Gateway sender is stopped. Event was not added to tmpDroppedEvents: {}"